tools/Customization/DevHome.Customization/ViewModels/FileExplorerViewModel.cs

```diff
@@ -187,4 +212,54 @@ await Task.Run(async () =>
         });
         RefreshTrackedRepositories();
     }
+
+    public bool CalculateEnabled(string settingName)
+    {
+        if (LocalSettingsService!.HasSettingAsync(settingName).Result)
+        {
+            return LocalSettingsService.ReadSettingAsync<bool>(settingName).Result;
+        }
+
+        // Settings disabled by default
+        return false;
+    }
+
+    [RelayCommand]
+    public async Task OnToggledVersionControlIntegrationSettingAsync()
+    {
+        IsVersionControlIntegrationEnabled = !IsVersionControlIntegrationEnabled;
+        await LocalSettingsService!.SaveSettingAsync("VersionControlIntegration", IsVersionControlIntegrationEnabled);
+
+        if (!IsVersionControlIntegrationEnabled)
+        {
+            _log.Information("The user has disabled version control integration inside Dev Home");
+            ExtraFolderPropertiesWrapper.UnregisterAllForCurrentApp();
+            _log.Information("Unregistered all repositories in File Explorer as setting is disabled");
+        }
+        else
+        {
+            _log.Information("The user has enabled version control integration in Dev Home.");
+            var repoCollection = RepoTracker.GetAllTrackedRepositories();
+            foreach (var repo in repoCollection)
+            {
+                ExtraFolderPropertiesWrapper.Register(repo.Key, typeof(SourceControlProvider).GUID);
+            }
+
+            _log.Information("Dev Home has restored registration for enhanced repositories it is aware about");
+        }
+    }
+
+    [RelayCommand]
+    public async Task OnToggledVersionControlInformationSettingAsync()
```

Comment:
How is this building?  This method and the method declared on line 228 have the same name, and don't differ by parameters.

Reply:
The names are different for the methods i.e. OnToggled**VersionControlIntegration**SettingAsync vs. OnToggled**VersionControlInformation**SettingAsync. I will consider if a better name can be used here. Thanks!